crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -799,6 +799,32 @@ where
                             return Some(Ok(()))
                         }
 
+                        // update the canon chain if continuous is enabled
+                        if self.sync.run_pipeline_continuously() {
+                            let max_block = ctrl.progress().unwrap_or_default();
+                            let max_hash = match self.blockchain.block_hash(max_block) {
+                                Ok(Some(block)) => block,
+                                Ok(None) => {
+                                    return Some(Err(Error::Provider(
+                                        ProviderError::HeaderNotFound(max_block.into()),
+                                    )
+                                    .into()))
+                                }
+                                Err(error) => return Some(Err(error.into())),
+                            };
+                            let max_header = match self.blockchain.block_by_number(max_block) {
```

Comment:
no need to look up the full block

Reply:
right, now this is a separate method which only fetches the header